libs/core/tests/unit_tests/caches/test_in_memory_cache.py

```diff
@@ -0,0 +1,125 @@
+from typing import Tuple
+
+import pytest
+
+from langchain_core.caches import RETURN_VAL_TYPE, InMemoryCache
+from langchain_core.outputs import Generation
+
+
+@pytest.fixture
+def cache() -> InMemoryCache:
+    """Fixture to provide an instance of InMemoryCache."""
+    return InMemoryCache()
+
+
+def cache_item(item_id: int) -> Tuple[str, str, RETURN_VAL_TYPE]:
+    """Generate a valid cache item."""
+    prompt = f"prompt{item_id}"
+    llm_string = f"llm_string{item_id}"
+    generations = [Generation(text=f"text{item_id}")]
+    return prompt, llm_string, generations
+
+
+def test_initialization() -> None:
+    """Test the initialization of InMemoryCache."""
+    cache = InMemoryCache()
+    assert cache._cache == {}
+    assert cache._maxsize is None
+
+    cache_with_maxsize = InMemoryCache(maxsize=2)
+    assert cache_with_maxsize._cache == {}
+    assert cache_with_maxsize._maxsize == 2
+
+    with pytest.raises(ValueError):
+        InMemoryCache(maxsize=0)
+
+
+def test_lookup(
+    cache: InMemoryCache,
+) -> None:
+    """Test the lookup method of InMemoryCache."""
+    prompt, llm_string, generations = cache_item(1)
+    cache.update(prompt, llm_string, generations)
+    assert cache.lookup(prompt, llm_string) == generations
+    assert cache.lookup("prompt2", "llm_string2") is None
+
+
+def test_update_with_no_maxsize(cache: InMemoryCache) -> None:
+    """Test the update method of InMemoryCache with no maximum size."""
+    prompt, llm_string, generations = cache_item(1)
+    cache.update(prompt, llm_string, generations)
+    assert cache.lookup(prompt, llm_string) == generations
+
+
+def test_update_with_maxsize() -> None:
+    """Test the update method of InMemoryCache with a maximum size."""
+    cache = InMemoryCache(maxsize=2)
+
+    prompt1, llm_string1, generations1 = cache_item(1)
+    cache.update(prompt1, llm_string1, generations1)
+    assert cache.lookup(prompt1, llm_string1) == generations1
+
+    prompt2, llm_string2, generations2 = cache_item(2)
+    cache.update(prompt2, llm_string2, generations2)
+    assert cache.lookup(prompt2, llm_string2) == generations2
+
+    prompt3, llm_string3, generations3 = cache_item(3)
+    cache.update(prompt3, llm_string3, generations3)
+
+    assert cache.lookup(prompt1, llm_string1) is None  # 'prompt1' should be evicted
+    assert cache.lookup(prompt2, llm_string2) == generations2
+    assert cache.lookup(prompt3, llm_string3) == generations3
+
+
+def test_clear(cache: InMemoryCache) -> None:
+    """Test the clear method of InMemoryCache."""
+    prompt, llm_string, generations = cache_item(1)
+    cache.update(prompt, llm_string, generations)
+    cache.clear()
+    assert cache.lookup(prompt, llm_string) is None
+
+
+@pytest.mark.asyncio
```

Comment:
We use auto-mode so, you don't need to type @pytest.mark.asyncio :)

Reply:
Oh nice, I removed it!